src/Psalm/Config.php

```diff
@@ -2107,6 +2107,16 @@ public function visitPreloadedStubFiles(Codebase $codebase, ?Progress $progress
             $core_generic_files[] = $stringable_path;
         }
 
+        if (PHP_VERSION_ID < 8_02_00 && $codebase->analysis_php_version_id >= 8_02_00) {
+            $stringable_path = dirname(__DIR__, 2) . '/stubs/Php82.phpstub';
+
+            if (!file_exists($stringable_path)) {
+                throw new UnexpectedValueException('Cannot locate PHP 8.2 classes');
+            }
+
+            $core_generic_files[] = $stringable_path;
+        }
```

Comment:
and `$stringable_path` is a misleading name (was used to import `Stringable` stub once and then was just copy-pasted around a lot).

Reply:
@orklah do you think we should drop `PHP_VERSION_ID` checks here? Stubs usually contain more detailed type info compared to reflection, so why don't we load them regardless of the runtime PHP version?

Reply:
I'd be very cautious here. The Reflection library change frequently and breaks often. For example, you don't want to say that some method can return `ReflectionIntersectionType` before PHP 8.2 because it will break all kind of assumptions. This specific case can be covered with conditional return, but the existence or not of some methods/class is more tricky I think

Reply:
Just be aware that I only replicated the 2 code blocks above this one

Reply:
To clarify, I don't want to drop `$codebase->analysis_php_version_id` check.

Now we have the following:
* `visitPreloadedStubFiles()` does this: 
  ```php
        if (PHP_VERSION_ID < 8_00_00 && $codebase->analysis_php_version_id >= 8_00_00) {
            // load Php80.phpstub
        }

        if (PHP_VERSION_ID < 8_01_00 && $codebase->analysis_php_version_id >= 8_01_00) {
            // load Php81.phpstub
        }
  ```
* `visitStubFiles()` does this:
  ```php
        if (PHP_VERSION_ID >= 8_00_00 && $codebase->analysis_php_version_id >= 8_00_00) {
             // load Php80.phpstub
        }

        if (PHP_VERSION_ID >= 8_01_00 && $codebase->analysis_php_version_id >= 8_01_00) {
             // load Php81.phpstub
        }
  ```
* Usually both are called, except when in the `diff` mode. When both are called, `PHP_VERSION_ID` check is useless (it will load stubs either in `visitPreloaded...` or in `visitStub...`.

Reply:
Based on local testing, I will indeed need to change this a bit.

Currently, the logic does:

```php
if (PHP_VERSION_ID < 8_00_00 && $codebase->analysis_php_version_id >= 8_00_00) {
```

Practically, it should just do this everywhere.

```php
if (PHP_VERSION_ID >= 8_00_00 && $codebase->analysis_php_version_id >= 8_00_00) {
```

Unsure about implications of changing this :thinking:

Reply:
Moved everything to `visitStubFiles`, and it looks fine :+1: